packages/clay-button/src/index.tsx

```diff
@@ -26,34 +31,44 @@ interface IProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
 	 */
 	monospaced?: boolean;
 
+	/**
+	 * Flag to indicate if link need have an outline.
+	 */
+	outline?: boolean;
+
 	/**
 	 * Indicates button should be a small variant.
 	 */
 	small?: boolean;
 }
 
 interface IClayButton extends React.FunctionComponent<IProps> {
-	Group: React.FunctionComponent<IButtonGroupProps>;
+	Group: typeof ClayButtonGroup;
 }
 
 const ClayButton: IClayButton = ({
 	block,
+	borderless,
 	children,
 	className,
 	displayType = 'primary',
 	monospaced,
+	outline,
 	small,
 	type = 'button',
 	...otherProps
 }: IProps) => (
 	<button
-		{...otherProps}
-		className={classNames(className, 'btn', `btn-${displayType}`, {
+		className={classNames(className, 'btn', {
 			'btn-block': block,
 			'btn-monospaced': monospaced,
+			'btn-outline-borderless': borderless,
 			'btn-sm': small,
+			[`btn-${displayType}`]: displayType && !outline,
+			[`btn-outline-${displayType}`]: displayType && outline,
```

Comment:
For `borderless` to have effect it needs `btn-outline-*`and should disable `btn-${displayType}`.
```suggestion
			[`btn-outline-${displayType}`]: displayType && (outline || borderless),
```